### PR TITLE
Add new checkers ``use-list-literal`` and ``use-dict-literal``

### LINKED ISSUE #4365: New checks: Use list / dict literal syntax
Sometimes I see stuff like this:

```python
x = list()
y = dict()
```

These would clearer as:

```python
x = []
y = {}
```

I would like to see two new checks, `use-list-literal` and `use-dict-literal`, or something like that.

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

Add ``use-list-literal`` and ``use-dict-literal`` checkers as described in #4365.

This adds two checks for when empty lists and dicts are created using a function call instead of their literals.
Tests and `pylint` code have been updated as well to conform to or ignore the new checker.

This closes #4365

### CODE REVIEW

**Comment by DanielNoord** [comment]
Little update, my IDE's automatic linter is a bit too aggressive.

Should be good now!

**Review by cdce8p — CHANGES_REQUESTED** [review]
I like this one!

File: tests/functional/u/use/use_dict_literal.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# pylint: disable=missing-docstring, invalid-name
+
+x = dict()  # [use-dict-literal]
```
Comment (cdce8p) [inline tests/functional/u/use/use_dict_literal.py:3]: Can you rename the test files?
```
use_dict_literal -> use_literal_dict
use_list_literal -> use_literal_list
```

That will make testing them a bit easier, as you then can do `pytest tests/test_functional -k use_literal` to just test these two.

Comment (Pierre-Sassoulas) [reply tests/functional/u/use/use_dict_literal.py:3]: Maybe ``-k literal`` is close enough ? :)

**Review by Pierre-Sassoulas — APPROVED** [review]
I was going to talk about ``safe_infer`` too but I think @cdce8p made a very nice point performance wise regarding the possibility to not have to infer every Call node.

File: tests/functional/u/use/use_dict_literal.py (hunk 1 above)
Comment (DanielNoord) [reply tests/functional/u/use/use_dict_literal.py]: I only renamed the test files. Should I have changed the message code as well? To keep it in line with document names? Or is this fine?

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 2]
```diff
@@ -1454,6 +1469,16 @@ def _check_consider_using_with(self, node: astroid.Call):
...
             self.add_message("consider-using-with", node=node)
 
+    def _check_use_list_or_dict_literal(self, node: astroid.Call) -> None:
+        """Check if empty list or dict is created by using the literal [] or {}"""
+        if node.as_string() in ("list()", "dict()"):
+            inferred = utils.safe_infer(node.func)
+            if isinstance(inferred, astroid.ClassDef) and not node.args:
+                if inferred.qname() == "builtins.list":
+                    self.add_message("use-list-literal", node=node)
+                if inferred.qname() == "builtins.dict" and not node.keywords:
```
Comment (cdce8p) [inline pylint/checkers/refactoring/refactoring_checker.py:1479]:
```suggestion
                elif inferred.qname() == "builtins.dict" and not node.keywords:
```

File: ChangeLog [hunk 3]
```diff
@@ -36,11 +36,20 @@ Release date: TBA
...
+* Added ``use-dict-literal``: Emitted when ``dict()`` is called with no arguments instead of using ``{}``
+
+  Closes #4365
+
 * ``CodeStyleChecker``
 
   * Extended ``consider-using-tuple`` check to cover ``in`` comparisons.
 
 
+
```
Comment (cdce8p) [inline ChangeLog:52]:
```suggestion
```

Empty line :)

**Review by cdce8p — COMMENTED** [review]
Just nitpicking at this point 😄

**Comment by DanielNoord** [comment]
> Just nitpicking at this point 😄

Isn't that the whole point of `pylint`? 😉

**Review by cdce8p — APPROVED** [review]

**Comment by cdce8p** [comment]
> > Just nitpicking at this point 😄
> 
> Isn't that the whole point of `pylint`? 😉

@DanielNoord It sometimes certainly feels that way (:
Thanks again 🐬

File: tests/functional/u/use/use_dict_literal.py (hunk 1 above)
Comment (DanielNoord) [reply tests/functional/u/use/use_dict_literal.py]: @cdce8p Just making sure you didn't miss this comment. No changes needed?

Comment (cdce8p) [reply tests/functional/u/use/use_dict_literal.py]: The comment was just about the test file names. So everything is fine with it

**Pull request merged**